js/oanda.js

```diff
@@ -0,0 +1,1632 @@
+        //            { price: '1.14508', liquidity: '1000000' },
+        //            { price: '1.14507', liquidity: '2000000' },
+        //            { price: '1.14506', liquidity: '2000000' },
+        //            { price: '1.14504', liquidity: '5000000' }
+        //         ],
+        //         asks: [
+        //            { price: '1.14525', liquidity: '1000000' },
+        //            { price: '1.14527', liquidity: '2000000' },
+        //            { price: '1.14528', liquidity: '2000000' },
+        //            { price: '1.14529', liquidity: '5000000' }
+        //         ],
+        //         closeoutBid: '1.14504',
+        //         closeoutAsk: '1.14529',
+        //         status: 'non-tradeable',
+        //         tradeable: false,
+        //         quoteHomeConversionFactors: { positiveUnits: '1.00000000', negativeUnits: '1.00000000' },
+        //         instrument: 'EUR_USD'
+        //       },
+        //       ...
+        //     ]
+        // }
+        const prices = this.safeValue (response, 'prices', []);
+        return this.parseTickers (prices, symbols, params);
+    }
+
+    parseTicker (ticker, market = undefined) {
+        const marketId = this.safeString (ticker, 'instrument');
+        market = this.safeMarket (marketId, market);
+        const symbol = this.safeSymbol (marketId, market);
+        // const status = this.safeString (ticker, 'status') === 'tradeable';
+        const date = this.safeString (ticker, 'time');
+        const timestamp = this.parseDate (date);
+        const bids = this.safeValue (ticker, 'bids');
+        const asks = this.safeValue (ticker, 'asks');
+        const bidBestObject = this.safeValue (bids, 0);
+        const askBestObject = this.safeValue (asks, 0);
+        const bidPrice = this.safeNumber (bidBestObject, 'price');
+        const bidVolume = this.safeNumber (bidBestObject, 'liquidity');
+        const askPrice = this.safeNumber (askBestObject, 'price');
+        const askVolume = this.safeNumber (askBestObject, 'liquidity');
+        return this.safeTicker ({
+            'symbol': symbol,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'high': undefined,
+            'low': undefined,
+            'bid': bidPrice,
+            'bidVolume': bidVolume,
+            'ask': askPrice,
+            'askVolume': askVolume,
+            'vwap': undefined,
+            'open': undefined,
+            'close': undefined,
+            'last': undefined,
+            'previousClose': undefined,
+            'change': undefined,
+            'percentage': this.safeString (ticker, 'percentChange'),
+            'average': undefined,
+            'baseVolume': this.safeString (ticker, 'baseVolume'),
+            'quoteVolume': this.safeString (ticker, 'quoteVolume'),
+            'info': ticker,
+        }, market, false);
+    }
+
+    async getListOfExecutions (symbols = undefined, params = {}) {
```

Comment:
Isn't is fetchMyTrades ?

Reply:
I've restructured them. You might not remind, Oanda had very ambigious endpoints for  '[trades](https://developer.oanda.com/rest-live-v20/trade-ep/)' and '[positions](https://developer.oanda.com/rest-live-v20/position-ep/)' , and after many checks, I've decided to use two different ways (can be overriden by 'method' param) to fetch trades from `fetchMyTrades` (you can check implementation of it).